ibis/backends/duckdb/__init__.py

```diff
@@ -556,11 +558,83 @@ def to_pyarrow(
         else:
             raise ValueError
 
-    def fetch_from_cursor(
+    @util.experimental
+    def to_parquet(
+        self,
+        expr: ir.Table,
+        path: str | Path,
+        *,
+        params: Mapping[ir.Scalar, Any] | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Write the results of executing the given expression to a parquet file.
+
+        This method is eager and will execute the associated expression
+        immediately.
+
+        Parameters
+        ----------
+        expr
+            The ibis expression to execute and persist to parquet.
+        path
+            The data source. A string or Path to the parquet file.
+        params
+            Mapping of scalar parameter expressions to value.
+        kwargs
+            DuckDB Parquet writer arguments. See
+            https://duckdb.org/docs/data/parquet#writing-to-parquet-files for
+            details
+        """
+        import ibis
+
+        query = str(ibis.to_sql(expr, dialect="duckdb", params=params))
```

Comment:
I am not using `self.compile()` explicitly because I am embedding this query into a string. I have to go through the the rigmarole of `compile_kwargs` etc if I use `self.compile()` and `to_sql` does that for me.

Reply:
I'm not sure I'm following. There is even a `_to_sql()` base backend method.